src/formatting-model-api.d.ts

```diff
@@ -0,0 +1,582 @@
+            /** defaulted to true, if false, we will keep the group enabled even though it's parent card has topLevelToggle value set to false */
+            inheritDisabled?: boolean;
+        }
+
+        export interface FormattingContainer extends VisualFormattingUid, VisualFormattingDisplayName {
+            containerItems: FormattingContainerItem[];
+            /**
+             * Whether this container allows editing, including add/remove container items, and
+             * edit of indivisual container item's value itself.
+             */
+            isEditable?: boolean;
+        }
+
+        export interface FormattingContainerItem extends VisualFormattingUid {
+            displayName: string;
+            disabled?: boolean;
+            slices?: FormattingSlice[];
+            groups?: Omit<FormattingGroup, 'containers'>[];
+        }
+
+        export type FormattingSlice = SimpleVisualFormattingSlice | CompositeVisualFormattingSlice;
+
+        export interface BaseVisualFormattingSlice<TControl> extends VisualFormattingUid {
+            control: TControl;
+            disabled?: boolean;
+            suppressUndoRedoRegister?: boolean;
+        }
+
+        /** Slice display name is required for composite slices */
+        export interface CompositeVisualFormattingSlice extends BaseVisualFormattingSlice<FormattingCompositeControl>, VisualFormattingDisplayName, VisualFormattingInfoIconText {
+        }
+
+        /** Slice display name is optional for simple slices */
+        export interface SimpleVisualFormattingSlice extends BaseVisualFormattingSlice<FormattingSimpleControl>, Partial<VisualFormattingDisplayName>, VisualFormattingInfoIconText {
+            // Properties added here to add comments
+            /** Only set if needed to override the capabilities */
+            displayName?: string;
+            /** Only set if needed to override the capabilities */
+            description?: string;
+        }
+
+        export interface EnabledSlice extends BaseVisualFormattingSlice<FormattingControls<Pick<SimpleComponentType, "ToggleSwitch">>> {
+            suppressDisplayName: boolean;
+        }
+
+        export interface VisualFormattingUid {
+            /** A unique Identifier */
+            uid: string;
+        }
+
+        export interface VisualFormattingDisplayName {
+            displayName: string;
+            description?: string;
+            /**
+             * It will be used as an alternate name for the formatting card (will be only used for searching purposes)
+             */
+            aliasName?: string;
+            suppressDisplayName?: boolean;
+        }
+
+        export interface VisualFormattingInfoIconText {
+            /** Only set if needed to display info icon with tooltip for the slice */
+            infoIconText?: string;
+        }
+
+        export interface FormattingDescriptor extends DataViewObjectPropertyIdentifier {
+            selector?: data.Selector;
```

Comment:
pls make sure selector is an OpaqueIdentity and no more

Reply:
data.Selector here is an refernece to empty interface, Its also the same in `VisualObjectInstanceEnumeration` which is used in `enumerateObjectInstances` and the search for `OpaqueIdentity` is in PowerbiClients